### PR TITLE
Story #25 task #114 & #149   home page and login card components

### PR DESCRIPTION
**Pull request by ZedTT**
Reviewing this once myself before asking someone else to. Other people have done work on this branch so this makes sense.

### CODE REVIEW

File: ReStyle-Backend/.gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 # Mac ignore
 /build
-/node_modules
\ No newline at end of file
+/node_modules
+/src/dbKey.ts
```
Comment (ZedTT) [inline ReStyle-Backend/.gitignore]: New line

File: ReStyle-Frontend/src/app/components/home-page/home-page.component.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<div style="margin: 65px 0;">
+  <app-login-card></app-login-card>
+  <app-item-card-stack></app-item-card-stack>
+</div>
```
Comment (ZedTT) [inline ReStyle-Frontend/src/app/components/home-page/home-page.component.html]: End in new line

File: ReStyle-Frontend/src/app/components/item-card-stack/item-card-stack.component.sass [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.stack
+    max-width: 700px
+    width: 80%
+    margin: 20px auto
```
Comment (ZedTT) [inline ReStyle-Frontend/src/app/components/item-card-stack/item-card-stack.component.sass]: End in new line

File: ReStyle-Frontend/src/app/components/login-card/login-card.component.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      </div>
+    </div>
+    <div class="card" style="display: flex">
+      <div style="width: 50%; height: 300px; border: 2px solid black; text-align: center; align-self: center">Please Log In</div>
+      <div class="card-footer buttons d-flex justify-content-around">
+        <button class="btn btn-success">Login</button>
+      </div>
+    </div>
+</div>
+<!--  -->
```
Comment (ZedTT) [inline ReStyle-Frontend/src/app/components/login-card/login-card.component.html]: End in new line. Not an empty comment

File: ReStyle-Frontend/src/app/components/login-card/login-card.component.sass [hunk 5]
```diff
@@ -0,0 +1,2 @@
+.hidden
+    display: none
```
Comment (ZedTT) [inline ReStyle-Frontend/src/app/components/login-card/login-card.component.sass]: End in new line

File: ReStyle-Frontend/src/app/components/login/login.component.html [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<div style="margin-top: 70px">
+    <firebase-ui></firebase-ui>
+</div>
```
Comment (ZedTT) [inline ReStyle-Frontend/src/app/components/login/login.component.html]: End in new line

File: ReStyle-Frontend/src/app/app.module.ts [hunk 7]
```diff
@@ -9,6 +17,41 @@ import { ItemCardStackComponent } from './components/item-card-stack/item-card-s
...
+    firebase.auth.TwitterAuthProvider.PROVIDER_ID,
+    firebase.auth.GithubAuthProvider.PROVIDER_ID,
+    {
+      requireDisplayName: false,
+      provider: firebase.auth.EmailAuthProvider.PROVIDER_ID
+    },
+    firebase.auth.PhoneAuthProvider.PROVIDER_ID,
+    firebaseui.auth.AnonymousAuthProvider.PROVIDER_ID
+  ],
+  tosUrl: '<your-tos-link>',
```
Comment (HaejoonChoi) [inline ReStyle-Frontend/src/app/app.module.ts]: URL of homepage after login should be added later

File: ReStyle-Frontend/src/app/app.module.ts [hunk 8]
```diff
@@ -9,6 +17,41 @@ import { ItemCardStackComponent } from './components/item-card-stack/item-card-s
...
+  privacyPolicyUrl: '<your-privacyPolicyUrl-link>',
```
Comment (HaejoonChoi) [inline ReStyle-Frontend/src/app/app.module.ts]: Also Privacy Policy Url (I forgot to add them)
- Need dummy page for Privacy Policy as well

File: ReStyle-Frontend/src/app/components/login-card/login-card.component.html [hunk 9]
```diff
@@ -0,0 +1,40 @@
+<div [ngClass]="setClasses()">
+    <div class="card" style="width: 18rem;">
+      <div id="carouselExampleControls" class="carousel slide" data-ride="carousel">
```
Comment (ZedTT) [inline ReStyle-Frontend/src/app/components/login-card/login-card.component.html]: The carousel is unfinished and should not be merged with master

File: ReStyle-Frontend/src/index.html [hunk 10]
```diff
@@ -1,6 +1,8 @@
 <!doctype html>
 <html lang="en">
 <head>
+  <link type="text/css" rel="stylesheet" href="https://cdn.firebase.com/libs/firebaseui/3.0.0/firebaseui.css" />
```
Comment (ZedTT) [inline ReStyle-Frontend/src/index.html]: This should be changed eventually. See https://www.npmjs.com/package/firebaseui-angular#add-the-firebaseui-css-to-your-imports
For now this is fine

Comment (ZedTT) [reply ReStyle-Frontend/src/index.html]: A comment should be added here, though

**Pull request merged**